### DISCUSSION TITLE
Enum prefix lint: use an alternate prefix?

### DISCUSSION THREAD

**Discussion by jaskij** [bufbuild/buf#3163] [Ideas]
There is the existing `ENUM_VALUE_PREFIX` lint, which works just fine. However, sometimes, the enum name is simply *long*. Would it be possible to designate an alternate prefix?

Say, we have an `enum CurrentDevice` - that's a bit long to type every single time, especially if the values themselves are already long. I'd like to require that every value in the enum starts with the prefix `CDEV_`.

Is it currently possible, and if not, can it be implemented?

**Comment by emcfarlane** [comment]
Not currently, but custom lint rules is something being investigated.

**Comment by KinNeko-De** [comment]
The prefixes are important for C++. Otherwise you can not use more than one enum there.
I use C# and Golang as output code. And the code generation of this plugis cut of the enum name in the generated code.

So you only have to type it in the proto. And AI hopefully help soon if you feed it with enough enums that follow the style guide :D
